CommandLine.c

```diff
@@ -255,6 +259,33 @@ static CommandLineStatus parseArguments(int argc, char** argv, CommandLineSettin
             }
             free_and_xStrdup(&flags->commFilter, optarg);
             break;
+
+         case 'S':
+            assert(optarg);
+            if (optarg[0] == '\0') {
+               fprintf(stderr, "Error: state filter cannot be empty.\n");
+               return STATUS_ERROR_EXIT;
+            }
+
+            for (char* c = optarg; *c != '\0'; c++) {
+               bool valid = false;
+
+               for (ProcessState s = UNKNOWN; s <= SLEEPING; s++) {
```

Comment:
Allow a comma/space-separated list here?  Might make for a better user experience on the command line.

Just a "if (*c == ',' || isspace(*c)) continue;" in the outer loop should do it.